### PR TITLE
Visualize precison recall curves for plates 3 and 3 prime

### PR DESCRIPTION
**Pull request by MattsonCam**
In this pr, I create precision-recall curves for plates 3 and 3 prime using a logistic regression model. I did not include HET cells in this evaluation due to these cells being contaminated. Please let me know if you have any questions or suggestions.

### CODE REVIEW

File: 2.evaluate_models/log_reg_plates_cp_fs_data/plates_3_3p/evaluate_model.ipynb [hunk 1]
```diff
@@ -1,384 +1,434 @@
...
-  {
-   "cell_type": "code",
-   "execution_count": 1,
-   "metadata": {
-    "execution": {
-     "iopub.execute_input": "2023-07-06T16:03:05.328393Z",
-     "iopub.status.busy": "2023-07-06T16:03:05.328148Z",
-     "iopub.status.idle": "2023-07-06T16:03:05.535031Z",
-     "shell.execute_reply": "2023-07-06T16:03:05.534697Z"
+    "cells": [
```
Comment (d33bs) [inline 2.evaluate_models/log_reg_plates_cp_fs_data/plates_3_3p/evaluate_model.ipynb:2]: Consider adding a markdown cell to the notebook or a code comment within this cell to help describe the purpose and goal of the work found in the notebook.

File: 2.evaluate_models/log_reg_plates_cp_fs_data/plates_3_3p/precison_recall_curves/precision_recall_plate_3_3p.png
Comment (d33bs) [inline 2.evaluate_models/log_reg_plates_cp_fs_data/plates_3_3p/precison_recall_curves/precision_recall_plate_3_3p.png:1]: Minor suggestion: is there a way to use common color groups (like "reds", "blues") or symbols (perhaps dotted-line, and "regular" or "bolded" line) for the `not_shuffled` and `shuffled` groups? This may help to differentiate these visually. If there's one to highlight over the other, you could use "warmer" colors or "bolder" outlines (line width, etc) to raise the visual to the audience.

File: 2.evaluate_models/log_reg_plates_cp_fs_data/plates_3_3p/nbconverted/evaluate_model.py [hunk 2]
```diff
@@ -39,84 +38,170 @@
...
-    fig_out_path=f"{fig_out_path}/log_reg_plates_3_3p_cp_fs_data_performance_bar_plot.png",
-)
+data_split = defaultdict(list)
+split_column = "Metadata_split_name"
+
+for model, data_type in list(product((logreg_model, logreg_model_shuf), ("train", "test"))):
+
+    # Only use the shuffled training feature for calculating the shuffled model
+    # unshuffled test data is used to evaluate the shuffled model
+    if (model == logreg_model_shuf):
```
Comment (d33bs) [inline 2.evaluate_models/log_reg_plates_cp_fs_data/plates_3_3p/nbconverted/evaluate_model.py:151]: Consider removing the parenthetical encapsulation here. Doing this removes two characters and increases readability in a minor way.

File: 2.evaluate_models/log_reg_plates_cp_fs_data/plates_3_3p/nbconverted/evaluate_model.py [hunk 3]
```diff
@@ -39,84 +38,170 @@
...
+        shuffled = "shuffled"
+        if (data_type == "train"):
```
Comment (d33bs) [inline 2.evaluate_models/log_reg_plates_cp_fs_data/plates_3_3p/nbconverted/evaluate_model.py:153]: Consider removing the parenthetical encapsulation here. Doing this removes two characters and increases readability in a minor way.

File: 2.evaluate_models/log_reg_plates_cp_fs_data/plates_3_3p/nbconverted/evaluate_model.py [hunk 4]
```diff
@@ -39,84 +38,170 @@
...
+
+    # Split the data based on the comparison being made
+    Xdf = featdf.loc[platedf[split_column] == X_split_name]
+    ydf = platedf.loc[platedf[split_column] == f"y_{data_type}"]["Metadata_genotype"]
+    ydf = ydf.reindex(Xdf.index)
+
+    # Encode the genotype labels
+    y_val = le.transform(ydf)
+
+    # Predict probabilites from the X data
```
Comment (d33bs) [inline 2.evaluate_models/log_reg_plates_cp_fs_data/plates_3_3p/nbconverted/evaluate_model.py:169]:
```suggestion
    # Predict probabilities from the X data
```

File: 2.evaluate_models/log_reg_plates_cp_fs_data/plates_3_3p/precison_recall_curves/precision_recall_plate_3_3p.png
Comment (d33bs) [inline 2.evaluate_models/log_reg_plates_cp_fs_data/plates_3_3p/precison_recall_curves/precision_recall_plate_3_3p.png:1]: Consider increasing the font size of the chart title, axis titles, and axis labels for increased readability. Feel free to ignore this if it's unhelpful or undesirable based on where this chart might be used.

File: 2.evaluate_models/log_reg_plates_cp_fs_data/plates_3_3p/nbconverted/evaluate_model.py [hunk 5]
```diff
@@ -39,84 +38,170 @@
...
+# ## Define and create paths
 
 # In[3]:
 
 
-sys.path.append(f"{root_dir}/2.evaluate_models/log_reg_plates_cp_fs_data/utils")
-import log_reg_plates_cp_fs_data_evaluate_util as au
+# Input paths
+filename3 = "Plate_3_sc_norm_fs.parquet"
+filename3p = "Plate_3_prime_sc_norm_fs.parquet"
```
Comment (d33bs) [inline 2.evaluate_models/log_reg_plates_cp_fs_data/plates_3_3p/nbconverted/evaluate_model.py:48]: Could these strings be references below for `path3` and `path3p` to help avoid variables which are only used once?

File: 2.evaluate_models/log_reg_plates_cp_fs_data/plates_3_3p/nbconverted/evaluate_model.py [hunk 6]
```diff
@@ -1,28 +1,27 @@
...
+from sklearn.metrics import precision_recall_curve
 
 # ## Find the root of the git repo on the host system
 
 # In[2]:
 
 
 # Get the current working directory
-cwd = Path.cwd()
+cwd = pathlib.Path.cwd()
```
Comment (d33bs) [inline 2.evaluate_models/log_reg_plates_cp_fs_data/plates_3_3p/nbconverted/evaluate_model.py:24]: What are the expectations for the CWD in this notebook? I see some later work to navigate the path, but I'm wondering specifically if this notebook is best run within a specific context. If there is a "best case" consider outlining that in a code comment here.

File: 2.evaluate_models/log_reg_plates_cp_fs_data/plates_3_3p/nbconverted/evaluate_model.py [hunk 7]
```diff
@@ -39,84 +38,170 @@
...
+model_all_data_path = f"{root_dir}/1.train_models/log_reg_plates_cp_fs_data/plates_3_3p"
+
+path3 = plate_path / filename3
+path3p = plate_path / filename3p
+
+model_data_path = pathlib.Path(f"{model_all_data_path}/model_data")
+model_path = pathlib.Path(f"{model_all_data_path}/trained_models")
 
+# Output path
+fig_path = pathlib.Path("precison_recall_curves")
```
Comment (d33bs) [inline 2.evaluate_models/log_reg_plates_cp_fs_data/plates_3_3p/nbconverted/evaluate_model.py:61]: Would it make sense to call this path "figures" to make it reusable or follow similar patterns in other directories?

File: 2.evaluate_models/log_reg_plates_cp_fs_data/plates_3_3p/nbconverted/evaluate_model.py [hunk 8]
```diff
@@ -39,84 +38,170 @@
...
 
 # In[6]:
 
 
-testdf = au.make_preds(testdf, lr)
+# Set plate column:
+plate3df["Metadata_plate"] = "3"
+plate3pdf["Metadata_plate"] = "3p"
+
+common_columns = plate3df.columns.intersection(plate3pdf.columns).to_list()
```
Comment (d33bs) [inline 2.evaluate_models/log_reg_plates_cp_fs_data/plates_3_3p/nbconverted/evaluate_model.py:106]: Consider providing a code comment here about any columns which won't be used and are ignored from this intersection.

File: 2.evaluate_models/log_reg_plates_cp_fs_data/plates_3_3p/nbconverted/evaluate_model.py [hunk 9]
```diff
@@ -39,84 +38,170 @@
...
+plate3df = plate3df.loc[:, common_columns]
+plate3pdf = plate3pdf.loc[:, common_columns]
```
Comment (d33bs) [inline 2.evaluate_models/log_reg_plates_cp_fs_data/plates_3_3p/nbconverted/evaluate_model.py:108]: Could you make the `df.loc()` references here directly in the `pd.concat()` below? The difference here may be minor, but in large data scenarios it could be helpful to not retain a reference to data which is no longer needed.

File: 2.evaluate_models/log_reg_plates_cp_fs_data/plates_3_3p/nbconverted/evaluate_model.py [hunk 10]
```diff
@@ -39,84 +38,170 @@
...
 
+# Combine the plate dataframes:
+platedf = pd.concat([plate3df, plate3pdf], axis="rows")
```
Comment (d33bs) [inline 2.evaluate_models/log_reg_plates_cp_fs_data/plates_3_3p/nbconverted/evaluate_model.py:111]: Consider the addition of something which can help visualize the changes to the dataset after this line. For example, `platedf["Metadata_plate"].unique()` or something of this nature.

File: 2.evaluate_models/log_reg_plates_cp_fs_data/plates_3_3p/nbconverted/evaluate_model.py [hunk 11]
```diff
@@ -39,84 +38,170 @@
...
-# ## Resave the testdf with Predictions
+
+# # Process data
 
 # In[7]:
 
 
-dump(testdf, models_path / "testdf.joblib")
+# Use only WT and Null Genotypes
+platedf = platedf.loc[platedf[gene_column] != "HET"]
```
Comment (d33bs) [inline 2.evaluate_models/log_reg_plates_cp_fs_data/plates_3_3p/nbconverted/evaluate_model.py:120]: Similar to the above comment, is there a way you could visualize this in the notebook, for example `platedf[gene_column].unique()` or `platedf[gene_column].value_counts()`?

File: 2.evaluate_models/log_reg_plates_cp_fs_data/plates_3_3p/nbconverted/evaluate_model.py [hunk 12]
```diff
@@ -39,84 +38,170 @@
...
+le = load(model_data_path / "label_encoder_log_reg_cp_fs_data_plate_3_3p.joblib")
 
-models_path = Path(
-    f"{root_dir}/1.train_models/log_reg_plates_cp_fs_data/plates_3_3p/data"
-)
+# Load split data indices as dataframe
+splitdf = pd.read_csv(model_data_path / "data_split_indices_log_reg_cp_fs_data_plate_3_3p.tsv", sep="\t", index_col=0)
+
+# Load feature name data
+feat_names = load(model_data_path / "feature_names_log_reg_cp_fs_data_plate_3_3p.joblib")
```
Comment (d33bs) [inline 2.evaluate_models/log_reg_plates_cp_fs_data/plates_3_3p/nbconverted/evaluate_model.py:81]: What data type does the `load` call here return? Consider making this declarative by indicating a type hint for the variable, for ex `feat_names : <vartype> = load(...)`.

File: 2.evaluate_models/log_reg_plates_cp_fs_data/plates_3_3p/nbconverted/evaluate_model.py [hunk 13]
```diff
@@ -39,84 +38,170 @@
...
+
+data_splitdf["plot_group"] = data_splitdf["model_type"] + " (" + data_splitdf["split_type"] + ")"
+
+sns.lineplot(data=data_splitdf, x="recall", y="precision", hue="plot_group", errorbar=None, linewidth=2)
+
+plt.legend(loc="lower right", title="Model (data split)")
+plt.ylim(bottom=0.19, top=1.02)
+plt.xlabel("Recall")
+plt.ylabel("Precision")
+plt.title("Precision vs. Recall Plates 3 and 3 prime Genotype Classification")
```
Comment (d33bs) [inline 2.evaluate_models/log_reg_plates_cp_fs_data/plates_3_3p/nbconverted/evaluate_model.py:205]: Would it be important to somehow add the shuffled vs non-shuffled content to the title? This may not be important, depending on the context of how this chart is used.

File: 2.evaluate_models/log_reg_plates_cp_fs_data/plates_3_3p/nbconverted/evaluate_model.py [hunk 14]
```diff
@@ -39,84 +38,170 @@
...
+
+# # Plot precision-recall curves
+
+# In[11]:
+
+
+plt.figure(figsize=(12, 10))
+sns.set_style("whitegrid")
+
+data_splitdf["plot_group"] = data_splitdf["model_type"] + " (" + data_splitdf["split_type"] + ")"
```
Comment (d33bs) [inline 2.evaluate_models/log_reg_plates_cp_fs_data/plates_3_3p/nbconverted/evaluate_model.py:197]: Consider adding a comment here about whats happening and why.

**Review by d33bs — APPROVED** [review]
Nice work! Overall I thought things looked good and only had comments towards improvements. Please don't hesitate to let me know if you have any questions.

**Comment by MattsonCam** [comment]
Thanks for reviewing @d33bs, I'll keep these suggestions in mind for future releases. After investigating this analysis further, I don't think this is as beneficial with the latest analysis updates, so I will close this pr.
